src/lib/comms/sol-lwm2m.c

```diff
@@ -3008,9 +3008,103 @@ clear_bootstrap_ctx(struct sol_lwm2m_client *client)
     }
 }
 
+static int
+is_authorized(struct sol_lwm2m_client *client,
+    int64_t server_id, uint16_t obj_id,
+    int32_t instance_id, int64_t rights_needed)
+{
+    struct obj_ctx *obj_ctx;
+    struct obj_instance *obj_instance;
+    struct sol_lwm2m_resource res[2];
+    int r = 0;
+    int64_t default_acl = SOL_LWM2M_ACL_NONE;
+    uint16_t i;
+
+    //If only one server or Bootstrap Server ID, then full access rights
+    if (sol_ptr_vector_get_len(&client->connections) == 1 || server_id == UINT16_MAX) {
+        SOL_DBG("Full access rights granted. This is either a Bootstrap Server"
+            " or single-server scenario");
+        return 1;
+    }
+
+    obj_ctx = find_object_ctx_by_id(client, ACCESS_CONTROL_OBJECT_ID);
+    SOL_VECTOR_FOREACH_IDX (&obj_ctx->instances, obj_instance, i) {
+        r = read_resources(client, obj_ctx, obj_instance, res, 2,
+            ACCESS_CONTROL_OBJECT_OBJECT_RES_ID,
+            ACCESS_CONTROL_OBJECT_INSTANCE_RES_ID);
+        if (r < 0) {
+            SOL_WRN("Could not read Access Control Object's"
+                " [Object ID] and [Instance ID] resources\n");
+            goto exit;
```

Comment:
Yes... Since these two would have to be `goto`s anyway:
https://github.com/solettaproject/soletta/pull/2255/files#diff-e4058dc14a343a8aceb7210b323358a2R3065
https://github.com/solettaproject/soletta/pull/2255/files#diff-e4058dc14a343a8aceb7210b323358a2R3068

I went with the approach of keeping them all syntactically consistent.